### PR TITLE
docs(docker versioning): rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Use simpler words
- Add better explanations
- Explain who does what

## Context

Drive-by PR. I think this needed a rewrite. 😄 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/versioning/docker/readme.md [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 
 **What type of versioning is used?**
 
-It's pretty "wild west" for tagging and not always compliant with SemVer. Docker versioning in Renovate should do a best effort to accept and sort SemVer-like versions.
+Docker image authors can use whatever tag they want, it's a "wild west".
+Docker tags don't always follow SemVer.
+
+This means that Renovate can only try to accept and sort SemVer-like versions, this won't always work though.
+You may need to help Renovate and create your own custom rules for some Docker images.
+TODO: find link to section that explains creating own Docker versioning/rules.
```
Comment (HonkingGoose) [inline lib/modules/versioning/docker/readme.md:21]: I need some help to find the best place to link to. I guess people sometimes need to create their own `packageRules` to work around a badly versioned Docker image tag?

```json
{
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["badly-versioned-docker-image"],
      "versioning": "insert-the-correct-string-here"
    }
  ]
}
```

**Review by HonkingGoose — COMMENTED** [review]
This was a tough rewrite, I hope it's better than what we had before. I have marked this PR as "draft" because of this review comment. 😉

File: lib/modules/versioning/docker/readme.md [hunk 2]
```diff
@@ -1,19 +1,31 @@
-Docker doesn't really have _versioning_, instead it supports "tags" and these are usually used by Docker image authors as a form of versioning.
+Docker doesn't really have _versioning_, instead it has "tags".
```
Comment (rarkins) [inline lib/modules/versioning/docker/readme.md:1]:
```suggestion
Docker images don't really have _versions_, instead they have "tags".
```

File: lib/modules/versioning/docker/readme.md (hunk 1 above)
Comment (rarkins) [reply lib/modules/versioning/docker/readme.md:21]: Use `loose` versioning in your example

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**